### ISSUE TITLE
Issue with chunk dates

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1290]
In GitLab by @franra9 on May 7, 2024, 13:32

( This is a template for bug reports. Please fill in the relevant information and delete the rest. )

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

4.1.5

### Expid affected(If applicable)

a1ah (lumi_vm)

### Which task has issues? Where is the log(If applicable)

* **Full_name**: general (i.e. APP and OPA)

* **Log_Path:** /appl/AS/AUTOSUBMIT_DATA/a1ah/tmp/*.cmd

### Summary

Dates wrongly set in the chunks date information. e.g. affecting CHUNK_START_DATE variable.

### Steps to reproduce

copy `a1ah`, and inspect

### What is the current bug behavior?

The dates corresponding to the chunks apparently are not correctly parsed.

### What is the expected correct behavior?

Dates corresponding to chunk 2 should be 19900102 but I have 19900105 (and weirdly sometimes other dates)

### Relevant logs and/or screenshots(if applicable)

the main reads:

```
EXPERIMENT:
   DATELIST: 19900101 #Startdate
   MEMBERS: fc0
   CHUNKSIZEUNIT: day
   CHUNKSIZE: 1
   NUMCHUNKS: 2
   CALENDAR: standard
```
I expect then `a1ah_19900101_fc0_1_1_APP_ENERGY_ONSHORE.cmd` (first day, first chunk):

```
(...)
DATELIST=${10:-19900101}
MEMBERS=${11:-fc0}
CHUNK=${12:-1}
INI_DAY=${14:-01}
INI_MONTH=${15:-01}
INI_YEAR=${16:-1990}
OPA_OUT=${17:-}
CHUNK_START_DATE=${18:-19900101}
(...)
```
(this is correct)

but then `a1ah_19900101_fc0_2_1_APP_ENERGY_ONSHORE.cmd` (first day, second chunk) I don't get the date information coming from the second daily chunk (19900102) but rather another date:

```
(...)
DATELIST=${10:-19900101}
MEMBERS=${11:-fc0}
CHUNK=${12:-2}
INI_DAY=${14:-05}
INI_MONTH=${15:-01}
INI_YEAR=${16:-1990}
OPA_OUT=${17:-}
CHUNK_START_DATE=${18:-19900105}
MEMBER=${19:-fc0}
END_DAY=${20:-05}
END_MONTH=${21:-01}
END_YEAR=${22:-1990}
(...)
```

the template that I use is [this one](https://earth.bsc.es/gitlab/digital-twins/de_340/workflow/-/blob/main/templates/application.sh):

```
DATELIST=${10:-%EXPERIMENT.DATELIST%}
MEMBERS=${11:-%EXPERIMENT.MEMBERS%}
CHUNK=${12:-%CHUNK%}
INI_DAY=${14:-%CHUNK_START_DAY%}
INI_MONTH=${15:-%CHUNK_START_MONTH%}
INI_YEAR=${16:-%CHUNK_START_YEAR%}
OPA_OUT=${17:-%OPAREQUEST.1.out_filepath%}
CHUNK_START_DATE=${18:-%CHUNK_START_DATE%}
MEMBER=${19:-%MEMBER%}
END_DAY=${20:-%CHUNK_SECOND_TO_LAST_DAY%}
END_MONTH=${21:-%CHUNK_SECOND_TO_LAST_MONTH%}
END_YEAR=${22:-%CHUNK_SECOND_TO_LAST_YEAR%}

```

### Any other relevant information(if applicable)

No other user mentioned this issue or found that previously. Also, a similar issue occurs in both AS4.1.2 and AS4.1.5

()

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 7, 2024, 13:55

I guess this is a bug derived from the splits calendar code (4.1.2). I'll revise it; maybe some default value for splits is messing with the chunk calendar. ( as these jobs have splits )

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 7, 2024, 14:02

more info at #1177 https://earth.bsc.es/gitlab/es/autosubmit/-/merge_requests/405 https://earth.bsc.es/gitlab/es/autosubmit/-/merge_requests/407

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 7, 2024, 14:07

Thinking about it, the thing is that it may be a configuration issue since this is a split, the split calendar code is taken into account... So to reproduce the old behavior <4.1.2.. you may need to change some parameters to have the old values. 

You never tested the calendar splits, no? I'll look into this tomorrow after finishing #1190

**Comment by kinow** [comment]
In GitLab by @franra9 on May 7, 2024, 14:26

Hi @dbeltrankyl , I never tested the calendar feature in the splits so far. I will try with 4.1.1 for example, to see if the error is still there, to discard other possible sources for this error.

**Comment by kinow** [comment]
In GitLab by @franra9 on May 7, 2024, 14:40

**UPDATE**

with V4.1.2 it does work properly. V4.1.5 does not. If you whish I can try with other versions too.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 7, 2024, 14:41

Ah right, the splits were added in 4.1.3 

```
4.1.3 - Bug fixes
=================
- Added Leonardo support.
- Improved inspect command.
- Added a new option to inspect.
- Wrapper now admits placeholders. 
- Reworked the wrapper deadlock code and total _jobs code. And fixed issues with "on_submission" jobs and wrappers. 
- Fixed issues with create without -f and splits.
- Improved error clarity.
- Added RO-Crate.
- Added Calendar for splits.

```

**Comment by kinow** [comment]
In GitLab by @franra9 on May 7, 2024, 14:45

Thanks @dbeltrankyl ! I'll use in the meantime 4.1.2.

**Comment by kinow** [comment]
In GitLab by @franra9 on May 7, 2024, 15:45

Extra info:

Using 4.1.2 I got the same error again. What I did to solve it has been to copy an older experiment while being in V4.1.2 (all the issue above comes from an `autosubmit --copy` while being in autosubmit 4.1.5).

In this way, I can run with 4.1.2.

**Comment by kinow** [comment]
In GitLab by @franra9 on May 8, 2024, 11:41

mentioned in issue digital-twins/de_340/workflow#358

**Comment by kinow** [comment]
In GitLab by @mcastril on May 8, 2024, 17:39

So, the issue happens when an older experiment is copied with a newer version?

CC @ainagaya

**Comment by kinow** [comment]
In GitLab by @franra9 on May 8, 2024, 17:45

The error happens when:

* whenever the app or end to edn wf is run with AS version >4.1.2

 AND
* if the experiment was copied from another experiment while having activated AS >4.1.2 even if then the new experiment is run with AS <=4.1.2

**Comment by kinow** [comment]
In GitLab by @mcastril on May 8, 2024, 19:08

Sorry @franra9 but I see a contradiction between the two premises. The first one says "run with AS > 4.1.2". The second says, "even if the new experiment is run with AS <=4.1.2"

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 9, 2024, 09:16

> if the experiment was copied from another experiment while having activated AS >4.1.2 even if then the new experiment is run with AS <=4.1.2

by copy do you mean expid --copy, or created + rerun without -f? If it is the second case it makes sense after debugging it.


I found and fixed the issue. 

The job.date was being updated with the dates from split to split. The thing is that this was a wrong approach as it should be another variable to don't interfere with the chunk dates as it was also looking at that variable. 

I've just added a self.date_split and now it is working fine on the branch

> If it is the second case it makes sense after debugging it.

If you swap from compatible versions in the same experiment, without touching the configuration, AS won't recreate the jobs, so the job.date was still incorrectly calculated.

**Comment by kinow** [comment]
In GitLab by @franra9 on May 9, 2024, 09:34

> by copy do you mean expid --copy, or created + rerun without -f? If it is the second case it makes sense after debugging it.

I can not exactly recall, I'm sorry.

Great @dbeltrankyl ! Thanks for the fast fixing.

hi @mcastril ,

I don't see a contradiction. Let me rephrase it so that it is more clear:

* When AS >4.1.2 --> Failed

* When AS = 4.1.2 AND previously AS >4.1.2 was used --> also failed

The second one may come as Dani says, from missing the -f flag

**Comment by kinow** [comment]
In GitLab by @mcastril on May 9, 2024, 10:13

I see @franra9, but then it should be an OR, not an AND (you cannot run with >4.1.2 and =4.1.2 at the same time. That's the contradiction I was seeing. In any case, I understand now. 

I think that somehow AS should detect if the syntax could be misinterpreted (by checking at the versions) and run with force if so...

**Comment by kinow** [comment]
In GitLab by @franra9 on Jun 10, 2024, 17:54

hi @dbeltrankyl ,

This issue was solved already in version 4.1.6 if I am not mistaken.
